testnet/stacks-node/src/neon_node.rs

```diff
@@ -178,12 +187,19 @@ fn inner_generate_poison_microblock_tx(
     keychain: &mut Keychain,
     nonce: u64,
     poison_payload: TransactionPayload,
+    is_mainnet: bool,
+    chain_id: u32,
 ) -> StacksTransaction {
     let mut tx_auth = keychain.get_transaction_auth().unwrap();
     tx_auth.set_origin_nonce(nonce);
 
-    let mut tx = StacksTransaction::new(TransactionVersion::Testnet, tx_auth, poison_payload);
-    tx.chain_id = TESTNET_CHAIN_ID;
+    let version = if is_mainnet {
```

Comment:
Maybe we should implement `From<bool>` for `TransactionVersion` -- it feels like we use this sort of `if/else` a lot.

Reply:
As in converting `mainnet` in `true` and `testnet` in `false`?
It feels like we'd been loosing in clarity.

Reply:
Yeah, you're right.  Let's leave it as-is.